src/data/serializer/Vector.js

```diff
@@ -184,14 +184,15 @@ Ext.define('GeoExt.data.serializer.Vector', {
         serialize: function(layer, source, viewRes) {
             var me = this;
             me.validateSource(source);
-            var features = source.getFeatures();
+            var map = Ext.ComponentQuery.query('gx_map')[0];
```

Comment:
Can we have the `map` as an argument to this function, so users can provide it themselves? Guessing the map is always error prone. And please keep the old behaviour as fallback if no map was given / found

Reply:
I changed the behaviour in 2ad05b51 so that all features are used in case no map is found. Unfortunately having the `map` as an argument is not really an option unless I rewrite the whole mapfish print provider and its serializer API. Can you live with this? I'm guessing that rewriting the print provider may also break other serializers not included in GeoExt.

Reply:
The newest iteration has the optional map argument in the `serialize`-method, and this is how I would want this to be.

Reply:
Yep, this should now be backwards compatible.